### PR TITLE
fix support for \!

### PR DESCRIPTION
**Pull request by mmatera**
This handles #1240

### CODE REVIEW

File: mathics/builtin/strings.py [hunk 1]
```diff
@@ -1688,6 +1695,31 @@ def apply_form(self, value, form, evaluation, options):
...
+      <dd>is the ad hoc fullform for \! $box$. just
+          for internal use...
+
+    >> \! \(2+2\)
+     = 4
+    </dl>
+    """
+
+    operator = "\\!"
+    precedence = 670
```
Comment (rocky) [inline mathics/builtin/strings.py:1711]: @mmatera Just curious - how did you arrive at the value 670?

**Comment by rocky** [comment]
@mmatera I don't have any thoughts or comments other than how the precidence was determined and how we can check that this is correct. 

If this helps, let's go with it.

File: mathics/builtin/strings.py (hunk 1 above)
Comment (mmatera) [reply mathics/builtin/strings.py:1711]: This seems to be the default in WL...
```
In[1]:= Precedence[anycustomsymbol]
Out[1]= 670
```

Comment (rocky) [reply mathics/builtin/strings.py:1711]: Thanks - for the information.

Comment (mmatera) [reply mathics/builtin/strings.py:1711]: Just a comment:  `InterpretedBox` is not a symbol in WL. However, someone else introduced it in the MathicsScanner (`\!`,  `\(`, etc) were already implemented in the scanner. I just hooked it to a new class that provides the basic elements needed for the parser, and a more or less simple way to reproduce the basic behavior.

Comment (rocky) [reply mathics/builtin/strings.py:1711]: Ok. As with too many other things, I guess we will address in a more complete fashion latter. 

For now this doesn't seem too disruptive or complex.

**Pull request merged**